src/shared/vscode/env.ts

```diff
@@ -95,3 +101,17 @@ export function getMinVscodeVersion(): string {
 export function getMinNodejsVersion(): string {
     return packageJson.devDependencies['@types/node'].replace(/[^~]/, '')
 }
+
+export async function copyUrl(window = Window.vscode(), env = Env.vscode(), url: string, onSuccess?: () => void) {
```

Comment:
signature isn't great.
could be further generalized for copy arn, etc.

Reply:
I was thinking of using this:
```ts
export async function copyToClipboard(data: string, label?: string): Promise<void> {
    await vscode.env.clipboard.writeText(data)
    const message = localize('AWS.explorerNode.copiedToClipboard', 'Copied {0} to clipboard', label)
    vscode.window.setStatusBarMessage(codicon`${getIcon('vscode-clippy')} ${message}`, 5000)
}
```

It uses stuff from the icon PR though

Reply:
shall i merge this and the icon PR can update the signature then?

Reply:
Yeah that'd work. May change it in another PR though, probably to remove `Window` and `Env`.

Reply:
You'll need to move the function somewhere else to avoid circular deps. Maybe `messages.ts`. Then I'd get rid of `Env.vscode()` and use `vscode.env` instead.

Reply:
cleaned up the signature and generalized it.

I left the env/window params at the end, since some tests are using that.